plugin/route53/route53.go

```diff
@@ -140,10 +143,72 @@ func (h *Route53) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg
 	return dns.RcodeSuccess, nil
 }
 
+const escapeSeq = `\\`
+
+// maybeUnescape parses s and converts escaped ASCII codepoints (in octal) back
+// to its ASCII representation.
+//
+// From AWS docs:
+//
+// "If the domain name includes any characters other than a to z, 0 to 9, -
+// (hyphen), or _ (underscore), Route 53 API actions return the characters as
+// escape codes."
+//
+// For our purposes (and with respect to RFC 1035), we'll fish for a-z, 0-9,
+// '-', '.' and '*' as the leftmost character (for wildcards) and throw error
+// for everything else.
+//
+// Example:
+//   `\\052.example.com.` -> `*.example.com`
```

Comment:
this should be supported by miekg/dns transparantly, i.e you can just carry this as-is and things should work. This is valid DNS presentation format.

Reply:
hm, i could've sworn I tried to pass this in a test first and it didn't work. Maybe just single `\`?

Reply:
Nope. Didn't work with neither `\\` nor `\` either when I short-circuit this func to just return argument.

Reply:
you're right this actually does the wrong thing:
~~~
package main

import "github.com/miekg/dns"

func main() {
	rr, _ := dns.NewRR(`\052.example.com. IN A 192.168.1.1`)
	println(rr.String())
}
~~~
*I think*